processing/src/main/java/io/druid/query/filter/BoundDimFilter.java

```diff
@@ -343,18 +345,33 @@ private void initPredicate()
             return;
           }
 
-          final boolean hasLowerLongBound;
-          final boolean hasUpperLongBound;
-          final long lowerLongBound;
-          final long upperLongBound;
+          boolean hasLowerLongBound;
+          boolean hasUpperLongBound;
+          long lowerLongBound;
+          long upperLongBound;
           boolean matchesNothing = false;
 
           if (hasLowerBound()) {
             final Long lowerLong = GuavaUtils.tryParseLong(lower);
             if (lowerLong == null) {
-              // Unparseable values fall before all actual numbers, so all numbers will match the lower bound.
-              hasLowerLongBound = false;
-              lowerLongBound = 0L;
+              BigDecimal lowerBigDecimal = getBigDecimalLowerBoundFromFloatString(lower);
+              if (lowerBigDecimal == null) {
+                // Unparseable values fall before all actual numbers, so all numbers will match the lower bound.
+                hasLowerLongBound = false;
+                lowerLongBound = 0L;
+              } else {
+                try {
+                  lowerLongBound = lowerBigDecimal.longValueExact();
+                  hasLowerLongBound = true;
+                } catch (ArithmeticException ae) { // the BigDecimal can't be contained in a long
```

Comment:
Does this mean that we won't check the lower bound because the ```lowerBigDecimal``` cannot be converted to a long value?
How about disallowing too large floating points for bound filters? I think it can be checked at the constructor of BoundDimFilter.